### ISSUE TITLE
[JAVA][SPRING-CLOUD] `spring-cloud` library generated code unusable with Java 8

### ISSUE THREAD

**Issue by wjam** [swagger-api/swagger-codegen#6681]
##### Description

As the interfaces generated when using the Spring library builds an interface with a default method when using Java 8, the Feign integration is rendered inoperable as Feign will run the default method rather than initiating the HTTP call.

##### Swagger-codegen version

2.2.3

##### Swagger declaration file content or url

See https://github.com/wjam/swagger-codegen-spring-cloud for an example

##### Command line used for generation

Maven library used. Configuration is in the GitHub repo mentioned above

##### Steps to reproduce

Run `mvn clean verify` on the GitHub repo mentioned above.

##### Suggest a fix/enhancement

Without considering any backwards incompatibility issues, I'd say drop the default method as it is not only useless but also has a dangerous default return value of `200` rather than doing something like throwing `java.lang.UnsupportedOperationException`.

Another alternative, less backwards incompatible way, would be to allow the control over whether the default methods are generated or not.

**Comment by wing328** [comment]
I think the latest master may have this issue addressed. Please give it a try with the latest master. The SNAPSHOT version can be found in the README.

**Comment by cbornet** [comment]
Yes, there's an issue : we shouldn't set the `jdk8` flag with spring-cloud.

**Comment by mumutu66** [comment]
@wing328  the snapshot solved the problem 
but if `dateLibrary` is set to `java8`  the `jdk8` still turn to true..semantically it's right but it will make the `spring-cloud-feign` library useless since the feign deals with interface default method in a special way,   so our need is even the `jdk8` is true , the generated feign client codes should not contain a default method within them
I sugguest a new template would be nice for spring-cloud-feign library
